src/ui/fileview.c

```diff
@@ -1379,27 +1379,30 @@ format_name(int id, const void *data, size_t buf_len, char buf[])
 	while(parent != child)
 	{
 		const char *prefix;
-		/* To avoid prepending, strings are reversed here and whole tree prefix is
-		 * reversed below to compensate for it. */
+		/* To avoid prepending, whole tree prefix is reversed below to compensate
+		 * for it. */
 		if(parent->child_count == child->child_pos + child->child_count)
 		{
-			prefix = (child == cdt->entry ? " --`" : "    ");
+			prefix = (child == cdt->entry ? "`-- " : "    ");
 		}
 		else
 		{
-			prefix = (child == cdt->entry ? " --|" : "   |");
+			prefix = (child == cdt->entry ? "|-- " : "|   ");
 		}
 		(void)sstrappend(buf, &len, buf_len + 1U, prefix);
 
 		child = parent;
 		parent -= parent->child_pos;
 	}
 
-	for(i = 0U; i < len/2U; ++i)
+	/* Reversing can be performed on chunks of four chars. */
+	uint32_t *left  = (uint32_t *)(buf),
+	         *right = (uint32_t *)(buf + len);
```

Comment:
This seems to be asking for a UB. C99, §6.3.2.3:

```
7 A pointer to an object or incomplete type may be converted to a pointer to a
  different object or incomplete type. If the resulting pointer is not correctly 
  aligned for the pointed-to type, the behavior is undefined. [...]
```
We can't guarantee that `buf` is properly aligned as it comes from the outside and branching based on whether alignment is correct seems to be overkill.

Reversing can be completely avoided by printing from right to left, but that would need figuring out depth of a node first, which might take more time than reversing (I haven't measured it though).

Reply:
It didn't catch my attention, thanks.

Hopefully, it can be fixed if you think it's worth it. Formatting functions only get called here, so adding alignment specifiers could solve it.

https://github.com/vifm/vifm/blob/4307041c509613d3fbca58d17b25469a7857ea08/src/ui/column_view.c#L280-L286

Reply:
I don't think that `column_view.c` should know about implementation details of formatters. The unit is meant to not have such dependencies.

I might have a better solution without reversing or using extra memory, but if we are talking about optimizations, we should at least try to measure their effect. Did you do any measurements yet?

Reply:
Sorry for not taking measures before. It only means 3-4 milliseconds "speedup" for 10 thousands of lines (depth ~6-7), so it just doesn't worth the effort. Next time I will be much more careful.